tedana/metrics/dependence.py

```diff
@@ -0,0 +1,354 @@
+"""
+Fit models.
+"""
+import logging
+import os.path as op
+
+import numpy as np
+import pandas as pd
+from scipy import stats
+
+from tedana import io, utils
+from tedana.stats import getfbounds, computefeats2, get_coeffs, t_to_z
+
+
+LGR = logging.getLogger(__name__)
+RepLGR = logging.getLogger('REPORT')
+RefLGR = logging.getLogger('REFERENCES')
+
+
+def calculate_weights(data_optcom, mixing_z):
+    """
+    Calculate weights.
+
+    Parameters
+    ----------
+    data_optcom : (S x T) array_like
+        Optimally combined data
+    mixing_z : (T x C) array_like
+        Z-scored mixing matrix
+
+    Returns
+    -------
+    weights : (S x C) array_like
+        Parameter estimates for optimally combined data against the mixing
+        matrix.
+    """
+    # compute un-normalized weight dataset (features)
+    weights = computefeats2(data_optcom, mixing_z, normalize=False)
+    return weights
+
+
+def calculate_betas(data_optcom, mixing):
+    assert data_optcom.shape[1] == mixing.shape[0]
```

Comment:
Can we make this a `ValueError` with a brief message instead of an assertion error?

Reply:
I'll add more informative checks (and tests) later on. I just wanted to throw a couple of checks in here as an example.